Vagrantfile

```diff
@@ -0,0 +1,66 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Use this Vagrantfile to run p2p-tests on an Intel macOS host.
+# This requires VirtualBox to be installed.
+#
+#   brew install vagrant
+#   vagrant up
+#   vagrant ssh
+#     cd /vagrant
+#     cargo build
+#     cd p2p-tests
+#     sudo -E ./contributor-fork-replication-1-test.ts
+
+Vagrant.configure("2") do |config|
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 8192
+    v.cpus = 4
+  end
+
+  config.vm.box = "ubuntu/focal64"
+  config.vm.provision "shell", inline: <<~SHELL
+    export DEBIAN_FRONTEND=noninteractive
+    apt-get -y update
+    apt-get -y upgrade
+    apt-get -y --no-install-recommends install \
+      build-essential \
+      ca-certificates \
+      cmake \
+      curl \
+      git \
+      jq \
+      procps; \
+    rm -rf /var/lib/apt/lists/*;
+
+    curl -sfLS https://deb.nodesource.com/setup_14.x | bash -; \
+    apt-get -y update; \
+    apt-get -y --no-install-recommends install nodejs; \
+    apt-get autoremove; \
+    rm -rf /var/lib/apt/lists/*; \
+    npm install --global yarn;
+
+    export RUST_VERSION=nightly-2021-12-14 \
+    export CARGO_HOME=/usr/local/cargo \
+    export PATH=/usr/local/cargo/bin:$PATH \
+    export RUSTUP_HOME=/usr/local/rustup \
+    export RUSTUP_VERSION=1.24.3 \
+    export RUSTUP_SHA256=3dc5ef50861ee18657f9db2eeb7392f9c2a6c95c90ab41e45ab4ca71476b4338
+
+    curl -sfLSO "https://static.rust-lang.org/rustup/archive/${RUSTUP_VERSION}/x86_64-unknown-linux-gnu/rustup-init"; \
+    echo "${RUSTUP_SHA256} *rustup-init" | sha256sum -c -; \
+    chmod +x rustup-init; \
+    ./rustup-init -y --no-modify-path --profile minimal --default-toolchain $RUST_VERSION; \
+    rm rustup-init; \
+    chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
+    rustup --version; \
+    cargo --version; \
+    rustc --version; \
+    rm -rf /usr/local/cargo/registry; \
+    rm -rf /usr/local/cargo/.package-cache;
+
+    echo "PATH=/usr/local/cargo/bin:$PATH" >> /home/vagrant/.bashrc
+    echo "export CARGO_HOME=/home/vagrant/cargo" >> /home/vagrant/.bashrc
+    echo "export CARGO_TARGET_DIR=/home/vagrant/cargo" >> /home/vagrant/.bashrc
```

Comment:
I don’t know if its a good idea to have them point to the same directory. Also not sure we need to set `CARGO_HOME`. The default value should be sufficient.

Reply:
351b3371f953c28460d53474b152be63bce9435f